File: problem1/solution/HandleHustle.cpp
#include <iostream>
#include <map>
#include <string>

using namespace std;

int main() {
    int n;
    cin >> n;
    
    map<string, int> usernameMap;
    
    for (int i = 0; i < n; ++i) {
        string username;
        cin >> username;
        
        if (usernameMap.find(username) == usernameMap.end()) {
            // Username not found, insert it with count 0
            usernameMap[username] = 0;
            cout << "OK" << endl;
        } else {
            // Username found, increment count and suggest new username
            ++usernameMap[username];
            cout << username << usernameMap[username] << endl;
        }
    }
    
    return 0;
}

File: problem1/solution/HandleHustle.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class UsernameRegistration {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        scanner.nextLine(); // Consume the newline character

        Map<String, Integer> usernameMap = new HashMap<>();

        for (int i = 0; i < n; ++i) {
            String username = scanner.nextLine();

            if (!usernameMap.containsKey(username)) {
                // Username not found, insert it with count 0
                usernameMap.put(username, 0);
                System.out.println("OK");
            } else {
                // Username found, increment count and suggest new username
                int count = usernameMap.get(username) + 1;
                usernameMap.put(username, count);
                System.out.println(username + count);
            }
        }

        scanner.close();
    }
}
